### DISCUSSION TITLE
main 서버 UT

### DISCUSSION THREAD

**Discussion by jminkkk** [woowacourse-teams/2024-code-zap#482] [이야기해봐요]

**Comment by jminkkk** [comment]
### 카피
- 템플릿 생성 시, 태그 설명 창이 잘 구분이 되지 않는다.
- 코드가 세로로 길 경우 한눈에 보기 어렵다 
  - 크면 어쩔 수 없는 것 같은데 지금은 좁은 느낌이다.
- 모든 태그 어라인

**Comment by jminkkk** [comment]
### 리브
- 테스트나 이전 코드 많이 재사용하게 된다
  - 깃허브를 많이 들어감
- 어떤 기능을 찾으려고 할 때 어떤 프로젝트에서 썼는지 헷갈릴 때가 있다
  - 깃허브에서 찾으려면 패키지를 타고타고 들어가야해서 불편하다
- 프리코스 때 코드 템플릿을 저장한 경험이 있다.
  - 깃허브의 별도 레포지토리를 파서 사용했다.
  - 가장 익숙해서 사용
- 깃허브에서 코드를 빠르게 찾기 위해 어떻게 할 수 있을까요?
  - 레포지토리 검색은 되는데 전체 내 레포지토리에서 특정 코드는 검색이 안되서 불편하다.
- 기존 코드잽 사용경험에서 추가로 있었으면 좋겠는 기능은?
  - 클래스를 통으로 기록할 것 같아서, 메서드 이름이나 클래스 명처럼 내용으로 검색할 수 있었으면 좋겠다.
- 정말 자주 쓰는 것은 검색하기도 귀찮을 것 같아서, 항상 맨위에 고정되어 있으면 좋겠다.

**Comment by kyum-q** [comment]
### 카피

- 레퍼런스는 저장해놓곤 합니다 (북마크)
- 북마스 사용이유?
	- 마우스 한번으로 쉽게 조작
	- 노션 정리는 찾기 어려움
- 찾기 힘든적?
	- 나는 많이 저장해놓지 않아서
	- 사소한 건 저장하지 않음
	- 양은 부담 X
- 재사용하고 싶은 코드를 찾을 때 뭘 먼저 떠올리는지? 왜 그걸 떠올렸나요?
	- 일단 하나하나 찾아봄
- 어떤 코드를 재사용하고 싶었나요?
	- 반복되는 코드 
	- 작성하는 사람의 아이디어가 필요한 것이 아닌 정해진 틀이 있는 코드
- 그 코드가 파일 하나인가요, 여러개인가요?
	- 여러개로 나눠져 있었다. 연관관계가 있다보니
- 원하는 코드 블럭을 찾고자 할 때 어떤 기능이 있으면 빠르게 찾을 수 있을 것 같나요?
	- 소제목
	- 설명
- 이런 서비스가 있을 때 어떤 점이 가장 큰 걸림돌이 될 거 같나요?
	- 코드가 많아지면 보기 힘들지 않을까?
- 저장 시에 북마크보다 시간이 걸린다. 이 부분에 대한 불편함은?
	- 북마크에도 한계가 있다. 
	- 북마크로 알기 쉬운것도 아님
	- 많아지면 관리가 어렵다
- 그렇게 저장한 코드들이 카테고리와 태그 등으로 분류가 될 거 같나요?
	- 될 것 같아요. 
- 유용한 기능

##### 사용성 테스트
- 카테고리 생성 설명이 너무 길다
- 태그 추가 칸 분리
- 설명 입력 칸 분리
- 태그 색상이 흐리다
- 여백이 너무 많다
- 코드 창 눌렀을 때 입력이 됐으면 좋겠다.
- 코드 가로 제한이 너무 작은 느낌?
- 모든 태그 글자가 넘친다

> 켬미는 모든 태그 기능이 있는지도 몰랐다.

**Comment by kyum-q** [comment]
### 리브


- 자신의 전 코드를 가져온 적 -> 많다
	- 간단한 CRUD 
	- 테스트 코드 작성 
	- 매번 동일한 코드를 작성해야할 때 Github를 보곤한다.
- Github를 방문하는 편인가?
- 어떻게 찾는가?
	- 어떤 레포였는지 기억하기가 어려웠다.
	- 다 들어가본다
- 별도로 저장해본 적이 있나?
	- 최종 코테 준비 시 저장했었다.
	- 깃허브 저장 레포를 파서 저장했다.
- 왜 깃허브
	- 제일 익숙해서
- 다른 튤 사용해봤나?
	- 사용했는데 코드는 깃허브가 편리할 것 같다고 생각
- 도움이 될만한 기능
	- 깃허브 <- 코드 검색이 안됨
- 찾고자하는게 안나오면 어땋게 하나?
	- 찾을 때까지 엄청 찾아봄
- 서비스에서 있었으면 좋겠는 것.
	- 자바 클래스를 통채로 올릴 것 같다
	- 그러니 클래스 이름으로 검색이 되면 좋을 것 같다.
	- 내용 검색이 안되는 것 같았다.
- 코드를 기록하고 조회할 때 가장 중요하다고 생각하는 것
	- 진짜 자주 쓰는 것은 고정할 수 있었으면 좋겠다.
